samples/cpp/speech_sample/main.cpp

```diff
@@ -286,161 +298,182 @@ int main(int argc, char* argv[]) {
-        // --------------------------- Step 5. Do inference --------------------------------------------------------
-        for (size_t next_output = 0; next_output < count_file; next_output++) {
-            std::vector<std::vector<uint8_t>> ptrUtterances;
-            std::vector<uint8_t> ptrScores;
-            std::vector<uint8_t> ptrReferenceScores;
-            ScoreErrorT frameError, totalError;
-            ptrUtterances.resize(inputFiles.size());
-            // initialize memory state before starting
-            for (auto&& state : inferRequests.begin()->inferRequest.query_state()) {
-                state.reset();
+        // --------------------------- Step 5. Do inference
+        // --------------------------------------------------------
+        std::vector<std::vector<uint8_t>> ptrUtterances;
+        std::vector<std::vector<uint8_t>> vectorPtrScores((outputs.size() == 0) ? 1 : outputs.size());
+        std::vector<uint8_t> ptrScores;
+        std::vector<uint16_t> numScoresPerOutput((outputs.size() == 0) ? 1 : outputs.size());
+        std::vector<uint8_t> ptrReferenceScores;
+        ScoreErrorT frameError, totalError;
+        ptrUtterances.resize(inputFiles.size());
+        // initialize memory state before starting
+        for (auto&& state : inferRequests.begin()->inferRequest.query_state()) {
+            state.reset();
+        }
+        /** Work with each utterance **/
+        for (uint32_t utteranceIndex = 0; utteranceIndex < numUtterances; ++utteranceIndex) {
+            std::map<std::string, ov::ProfilingInfo> utterancePerfMap;
+            uint64_t totalNumberOfRunsOnHw = 0;
+            std::string uttName;
+            uint32_t numFrames(0), n(0);
+            std::vector<uint32_t> numFrameElementsInput;
+            uint32_t numFramesReference(0), numFrameElementsReference(0), numBytesPerElementReference(0),
+                numBytesReferenceScoreThisUtterance(0);
+
+            /** Get information from input file for current utterance **/
+            numFrameElementsInput.resize(numInputFiles);
+            for (size_t i = 0; i < inputFiles.size(); i++) {
+                std::vector<uint8_t> ptrUtterance;
+                auto inputFilename = inputFiles[i].c_str();
+                uint32_t currentNumFrames(0), currentNumFrameElementsInput(0), currentNumBytesPerElementInput(0);
+                file->get_file_info(inputFilename, utteranceIndex, &n, &numBytesThisUtterance[i]);
+                ptrUtterance.resize(numBytesThisUtterance[i]);
+                file->load_file(inputFilename,
+                                utteranceIndex,
+                                uttName,
+                                ptrUtterance,
+                                &currentNumFrames,
+                                &currentNumFrameElementsInput,
+                                &currentNumBytesPerElementInput);
```

Comment:
It would be nice to introduce structure to work with files. The same things are doing several times for input/output/reference files. Too many variables which is hard to track.

For example:
```
        struct UtteranceInfo {
            uint32_t num_frames = 0;            // num_rows
            uint32_t num_frame_elements = 0;    // num_cols
            uint32_t num_bytes_per_element = 0; // elem_size
            uint32_t num_bytes = 0;             // num_rows * num_cols * elem_size
            std::vector<uint8_t> data = {};
            uint8_t* data_ptr = nullptr;
        };

        struct FileInfo {
            std::string path = "";
            BaseFile* file_ptr = nullptr;
            uint32_t num_utterances = 0;
            uint32_t num_memory_bytes = 0;
            std::vector<UtteranceInfo> u_info = {};
        };
```